### ISSUE TITLE
Adjust the behavior of `skip_result` custom rules

### ISSUE THREAD

**Issue by valeriy42** [elastic/ml-cpp#2553]
Currently, if we define `skip_result` it avoids updating distribution quantiles. This can change the impact on the anomaly score, although the anomaly probability remains the same. 

Ideally, we need to change the behavior such that for `skip_result`, the algorithm behaves "normally", except it avoids writing records and changing aggregates.

**Comment by valeriy42** [comment]
Maybe we should introduce `result.s_ShouldGetAnomaly` or something like that similar to `result.s_ShouldUpdateModel` to control this behavior explicitly.

**Comment by droberts195** [comment]
I don't think we should introduce a third user-facing config option. Updating the model but not the quantiles just sounds wrong if the rule was set up to update the model but not write out any anomalies.
